emulator/src/extensions/tools/emu_console_cmds.c

```diff
@@ -40,12 +38,22 @@ static const char* commandDocs[][3] = {
      "begin recording inputs into replay file [filename], or an auto-generated file name if not specified"},
     {"fuzz", "fuzz [on|off]", "toggles the fuzzer"},
     {"fuzz buttons", "fuzz buttons [on|off]", "toggles fuzzing of button inputs"},
-    {"fuzz buttons mask", "fuzz buttons mask [<button>[ <button>  ...]]",
+    {"fuzz buttons mask", "fuzz buttons mask [<button> [<button>  ...]]",
```

Comment:
When doing bigbug fuzzing, I tweaked it so that the buttons had different randomized weights, and the weights changed over time. Basically this would move Garbotnik in different directions rather than have him twitch in place forever. That doesn't feel feasible as CLI, but might be doable as JSON config? I'm not sure how valuable it would be, as the magic of fuzzing is that you usually do it without thinking too much...

Reply:
Ah, this is a pretty good idea. I think there actually was a little bit of configuration for setting probability on the 2023(?) emulator but yeah something a bit better than completely random button presses would be good. The other thing is it's _constantly_ pressing buttons but never for very long which yeah usually means not much happens. It wouldn't be too hard to make even a slightly smarter fuzzer. Even if it just spaced out button presses a bit and held them for more reasonable times that would be a big improvement.

And yeah some sort of emulator config might be a good idea, some of the parameters are getting a bit unwieldy and we've accumulated a good bit of stuff we'd actually want to configure.

Reply:
Yeah, turns out when you make features complex, usage goes down. It's a tough tradeoff.

The holy grail would be a fuzzer that had enough sense to "do something" rather than just hammer on buttons. I vaguely remember using an Android one at one point which would take screenshots and 'reward' the fuzzer for getting to different states or something like that.

Reply:
Oh hey https://llvm.org/docs/LibFuzzer.html

Reply:
> Oh hey https://llvm.org/docs/LibFuzzer.html
> Maybe we should not "roll our own" here

Oooh, wow! That's incredible, if I'm understanding that interface right then it should be trivial to hook that into the existing button input system. I might try that later.

Reply:
Yep, it looks trivial to hook up to buttons, but it does recommend having deterministic targets that don't modify global state, and we have neither. Even with seeded rng, imprecise us timers make it non deterministic

Reply:
Hmm... We do have the "fake timer" stuff that should help with that... it should give modes exactly `frameTimeUs` every frame, regardless of how long it actually took. More deterministic fuzzing is basically what it's for!